src/idlcxx/src/streamer_generator.c

```diff
@@ -1504,20 +1505,52 @@ idl_retcode_t process_case_label(context_t* ctx, idl_case_label_t* label)
     char* buffer = NULL;
     idl_constval_t* cv = (idl_constval_t*)ce;
 
-    if (idl_is_masked(ce, IDL_INTEGER_TYPE))
+    switch (ce->mask % (IDL_BASE_TYPE * 2))
```

Comment:
Please refrain from doing do this in the future. The mask is not a supported "interface". When changes occur these things will break. I'll allow this in for now as the next iteration will have `idl_type()`, but it would have made sense for the backend to add function for this itself.